client/src/pages/private/ParticipantDetailsView.js

```diff
@@ -62,21 +45,49 @@ export default () => {
   // State
   const [error, setError] = useState(null);
   const [participant, setParticipant] = useState(null);
+  const [actualParticipant, setActualParticipant] = useState(null);
+  const [showEditModal, setShowEditModal] = useState(false);
+  const [message, setMessage] = useState('Loading participant details');
+  // Auth context
+  const { auth } = AuthContext.useAuth();
+  // Memo roles
+  const roles = useMemo(() => auth.user?.roles || [], [auth.user?.roles]);
   // Style classes
   const classes = customStyle();
   // Get param
   const { id } = useParams();
+  // Edit Button flag
+  const enableEdit = roles.some((role) => ['ministry_of_health', 'superuser'].includes(role));
 
+  // UI Actions
+  // 1. Show edit
+  const showEditInfoModal = async () => setShowEditModal(true);
+  // 2. Update Info
+  const onUpdateInfo = async (values) => {
+    setShowEditModal(false);
+    try {
+      const [updatedParticipant] = await updateParticipant(values, actualParticipant);
+      setMessage('Update Success');
+      setParticipant(displayData(updatedParticipant));
+      setActualParticipant(updatedParticipant);
+      setTimeout(() => {
```

Comment:
Should we be using set timeout here? why are we introducing a delay, and why this specific value?

Reply:
You are right toast will be better idea